### PR TITLE
add entry type matcher for debit or credit entry

### PR DESCRIPTION
**Pull request by alovak**
I want to test the return of `credit` transfer. Currently you can add matcher only for `debit`. With this PR we can specify type on the entry in a matcher.

```yaml
  - match:
      entryType: credit
      amount:
        min: 100000 # $1,000
        min: 120000 # $1,200
    action:
      return:
        code: "R01"
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,7 +66,7 @@ match:
     min: <integer>
     max: <integer>
     value: <integer>    # Either min AND max OR value is used
-  debit: <object>       # Include this to only match on debits
+  type: <string>       # credit, debit or empty string to skip the type match
```
Comment (adamdecaf) [inline README.md:69]: I had intended there to be debit/credit as objects that allow for future expansion. I'm forgetting an example right now and don't see one in my notes. 

`type` is pretty generic. I think there'd be an `accountType: <string>` field.

Comment (alovak) [reply README.md:69]: Let's make it an `entryType` instead of `accountType`. As it's more about operation type than about account type.

TransactionCode we match against consists of two pieces of information:
* entry type: debit or credit
* account: checking, savings, etc.

Later we may add a more complex structure for this matcher. I think `entryType: debit/credit` is good enough for now.

Thoughts?

File: pkg/response/matcher.go [hunk 2]
```diff
@@ -82,12 +82,12 @@ func (m Matcher) FindAction(ed *ach.EntryDetail) *service.Action {
 		}
 
 		// Check if this Entry is a debit
-		if matcher.Debit != nil {
-			if matchedDebit(matcher, ed) {
-				m.debugLog(fmt.Sprintf("EntryDetail.TransactionCode=%d debit positive match", ed.TransactionCode))
+		if matcher.EntryType != service.EntryTypeEmpty {
+			if matchedType(matcher, ed) {
+				m.debugLog(fmt.Sprintf("EntryDetail.TransactionCode=%d type positive match", ed.TransactionCode))
```
Comment (adamdecaf) [inline pkg/response/matcher.go:87]:
```suggestion
			if matchedEntryType(matcher, ed) {
				m.debugLog(fmt.Sprintf("EntryDetail.TransactionCode=%d entryType positive match", ed.TransactionCode))
```

File: pkg/response/matcher.go [hunk 3]
```diff
@@ -149,7 +160,6 @@ func matchedIndividualName(m service.Match, ed *ach.EntryDetail) bool {
 }
 
 func (m Matcher) debugLog(msg string) {
-	if m.Debug {
-		m.Logger.Info().Log(msg)
-	}
+	fmt.Println(msg)
+	// m.Logger.Info().Log(msg)
```
Comment (adamdecaf) [inline pkg/response/matcher.go:164]: Can we undo this change?

File: pkg/response/matcher_test.go [hunk 4]
```diff
@@ -24,25 +24,30 @@ func TestMatchAccountNumber(t *testing.T) {
 	require.False(t, matchesAccountNumber(m, ed))
 }
 
-func TestMatchDebit(t *testing.T) {
-	m := service.Match{
-		Debit: &service.Debit{},
+func TestMatchType(t *testing.T) {
```
Comment (adamdecaf) [inline pkg/response/matcher_test.go:27]:
```suggestion
func TestMatchEntryType(t *testing.T) {
```

**Comment by alovak** [comment]
@adamdecaf I've fixed naming issues and addressed other mentioned issues.

**Review by adamdecaf — APPROVED** [review]
